File: Single_number.cpp
//Time Complexity-O(n)
//Space Complexity-O(1)
//Did the code execute on Leetcode? Yes

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        if(nums.size()==0)
        {
            return 0;
        }
        int x=nums[0];
        for(int i=1;i<nums.size();i++)
        {
            x=x^nums[i];
        }
        return x;
    }
};

File: src/SingleNumber.java
//Time Complexity : O(N)
//Space Complexity : O(1)
//Did this code successfully run on Leetcode : Yes
//Any problem you faced while coding this : No

class Solution {
	public int singleNumber(int[] nums) {
		int ans = 0;
		for (int num : nums)
			ans = ans ^ num;
		return ans;
	}
}
